src/Symfony/Component/Serializer/Normalizer/ConstraintViolationListNormalizer.php

```diff
@@ -34,19 +34,29 @@ public function normalize($object, $format = null, array $context = array())
         foreach ($object as $violation) {
             $violations[] = array(
                 'propertyPath' => $violation->getPropertyPath(),
-                'message' => $violation->getMessage(),
-                'code' => $violation->getCode(),
+                'type' => sprintf('urn:uuid:%s', $violation->getCode()),
```

Comment:
I don't think we should change it from "message" and "code" here, especially because:

>       "title" (string) - ... It SHOULD NOT change from occurrence to occurrence of the
>       problem, except for purposes of localization (e.g., using
>       proactive content negotiation

Reply:
It's ok. For the same problem, the same message and the same code will always be the same. They share the same semantic at a different level (for a given constraint violation, the code is always the same).

(Btw, the spec applies only to the root keys, not to our extension).

Reply:
> For the same problem, the same message and the same code will always be the same.

The message is customizable, so it might not be the same.

> (Btw, the spec applies only to the root keys, not to our extension).

Yes, that's also why I don't see why we should try to mimic the spec here... It only creates more confusion.

Reply:
> The message is customizable, so it might not be the same.
The good practice will be to change the code if you change the message. It's not very easy right now, but we can improve this in the validator later.

Anyway it's just a `SHOULD NOT`:

> This phrase, or the phrase "NOT RECOMMENDED" mean that
   there may exist valid reasons in particular circumstances when the
   particular behavior is acceptable or even useful, but the full
   implications should be understood and the case carefully weighed
   before implementing any behavior described with this label.

It's up to the developper to change the UUID if it uses this normalizer and change the default message and want to follow the spec's best practice (not very common).

> Yes, that's also why I don't see why we should try to mimic the spec here... It only creates more confusion.

Consistency. It's very confusing to use different terms for the same usage.

Reply:
> Consistency. It's very confusing to use different terms for the same usage.

It's not the same usage... We don't have to think of it like that.

Reply:
In the context of constraint violations, it's very obvious what code and message means. It's instantly familiar to Symfony users. Adopting the RFC 7807 semantics for this makes no sense.